components/webhook-mgt/org.wso2.carbon.identity.webhook.metadata/src/main/java/org/wso2/carbon/identity/webhook/metadata/api/service/WebhookMetadataService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.webhook.metadata.api.service;
+
+import org.wso2.carbon.identity.webhook.metadata.api.exception.WebhookMetadataException;
+import org.wso2.carbon.identity.webhook.metadata.api.model.EventProfile;
+import org.wso2.carbon.identity.webhook.metadata.api.model.ProfileType;
+
+/**
+ * Service interface for webhook metadata.
+ * This interface defines methods to get webhook event metadata.
+ */
+public interface WebhookMetadataService {
+
+    /**
+     * Get the event profile metadata for a specific profile type.
+     *
+     * @param profileType Type of event profile
+     * @return EventProfile containing channel and event metadata
+     * @throws WebhookMetadataException If an error occurs while retrieving the event profile
+     */
+    EventProfile getEventProfile(ProfileType profileType) throws WebhookMetadataException;
```

Comment:
Shouldn't there be a method to get supported profile types

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6782/commits/18d9e9a65adb6dc98ea02b0021b59c5d0804bcaf